include/vcpkg/base/util.h

```diff
@@ -107,8 +106,8 @@ namespace vcpkg::Util
         }
     }
 
-    template<class Range, class Pred, class E = ElementT<Range>>
-    std::vector<E> filter(const Range& xs, Pred&& f)
+    template<class Range, class Pred, class E = ElementT<const Range&>>
+    std::vector<E> filter(const Range& xs, Pred f)
```

Comment:
If I recall correctly, I wrote this and my logic in having it be a template parameter was to make it overridable at the caller -- the idea was that the caller could say "Hey, I actually wanted a vector of `unsigned` instead of `int`" or such.

I suppose this argues that we should consider a ranges library (though I'm concerned about compile time perf).

Reply:
If you wanted the caller to override it then it would probably need to be the first template parameter rather than the last one so the caller wouldn't have to specify deduced parameters though.